scene/2d/skeleton_2d.cpp

```diff
@@ -77,9 +171,200 @@ void Bone2D::_notification(int p_what) {
 			skeleton = nullptr;
 		}
 		parent_bone = nullptr;
+		set_transform(cache_transform);
+	}
+
+	else if (p_what == NOTIFICATION_READY) {
+		if (autocalculate_length_and_angle) {
+			calculate_length_and_rotation();
+		}
+	}
+#ifdef TOOLS_ENABLED
+	else if (p_what == NOTIFICATION_EDITOR_PRE_SAVE || p_what == NOTIFICATION_EDITOR_POST_SAVE) {
+		Transform2D tmp_trans = get_transform();
+		set_transform(cache_transform);
+		cache_transform = tmp_trans;
+	}
+	// Bone2D Editor gizmo drawing:
+#ifdef _MSC_VER
```

Comment:
This needs to be `#ifndef _MSC_VER` instead of `#ifdef _MSC_VER`. Windows compilation is failing.